vpr/src/draw/draw.cpp

```diff
@@ -3997,6 +4066,577 @@ void net_max_fanout(GtkWidget* /*widget*/, gint /*response_id*/, gpointer /*data
+    std::string expr = gtk_entry_get_text((GtkEntry*)entry);
+    if (valid_expression(expr)) {
+        draw_state->list_of_breakpoints.push_back(breakpoint("bt_expression", expr));
+        std::string bpDescription = "Breakpoint at " + expr;
+        add_to_bpList(bpDescription);
+    } else
+        invalid_entry_window("Invalid expression");
+}
+
+//window that pops up when an entry is not valid
+void invalid_entry_window(std::string error) {
+    //window settings
+    GtkWidget* window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_position((GtkWindow*)window, GTK_WIN_POS_CENTER);
+    gtk_window_set_title((GtkWindow*)window, "ERROR");
+    gtk_window_set_modal((GtkWindow*)window, TRUE);
+
+    //grid settings
+    GtkWidget* grid = gtk_grid_new();
+
+    //label settings
+    GtkWidget* label = gtk_label_new(error.c_str());
+    gtk_widget_set_margin_left(label, 30);
+    gtk_widget_set_margin_right(label, 30);
+    gtk_widget_set_margin_top(label, 30);
+    gtk_widget_set_margin_bottom(label, 30);
+    gtk_grid_attach((GtkGrid*)grid, label, 0, 0, 1, 1);
+
+    //button settings
+    GtkWidget* button = gtk_button_new_with_label("OK");
+    gtk_widget_set_margin_bottom(button, 30);
+    gtk_widget_set_margin_right(button, 30);
+    gtk_widget_set_margin_left(button, 30);
+    gtk_grid_attach((GtkGrid*)grid, button, 0, 1, 1, 1);
+    g_signal_connect(button, "clicked", G_CALLBACK(ok_close_window), window);
+
+    gtk_container_add(GTK_CONTAINER(window), grid);
+    gtk_widget_show_all(window);
+}
+
+//window that pops up when a breakpoint is reached
+//shows which breakpoint the program has stopped at and gives an info summary
+void breakpoint_info_window(std::string bpDescription, current_information current_info_d) {
+    //window settings
+    GtkWidget* window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_position((GtkWindow*)window, GTK_WIN_POS_CENTER);
+    gtk_window_set_title((GtkWindow*)window, "Breakpoint");
+
+    //grid settings
+    GtkWidget* grid = gtk_grid_new();
+
+    //label settings
+    GtkWidget* label = gtk_label_new(bpDescription.c_str());
+    gtk_widget_set_margin_left(label, 30);
+    gtk_widget_set_margin_right(label, 30);
+    gtk_widget_set_margin_top(label, 30);
+    gtk_widget_set_margin_bottom(label, 30);
+    gtk_grid_attach((GtkGrid*)grid, label, 0, 0, 1, 1);
+
+    GtkWidget* curr_info = gtk_label_new(NULL);
+    gtk_label_set_markup((GtkLabel*)curr_info, "<b>Current Information</b>");
+    gtk_widget_set_margin_left(curr_info, 30);
+    gtk_widget_set_margin_right(curr_info, 30);
+    gtk_widget_set_margin_bottom(curr_info, 15);
+    gtk_grid_attach((GtkGrid*)grid, curr_info, 0, 1, 1, 1);
+
+    //info grid
+    GtkWidget* info_grid = gtk_grid_new();
+    gtk_widget_set_margin_left(info_grid, 30);
+    gtk_widget_set_margin_right(info_grid, 30);
+    gtk_widget_set_margin_bottom(info_grid, 20);
+
+    //images
+    GtkWidget* m = gtk_image_new_from_file("src/draw/m.png");
```

Comment:
Pretty short names.  Should comment what they are and more descriptive file names would be better.

Reply:
the images are icons of the letters. So it's literally an image of the letter m. It's just something I used to make the information pop up look nicer and less wordy. Should I still change the names?

Reply:
OK, in that case the names are OK.  Should still comment what they are (these are just letters).  
Could keep the current names, or change them to letter_m.png and such.

Reply:
added the comment and resolved